tests/regression/kernel/test_callstack

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+#
+# Copyright (C) - 2017 Francis Deslauriers <francis.deslauriers@efficios.com>
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License, version 2 only, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+TEST_DESC="Kernel tracer - Callstack context"
+
+CURDIR=$(dirname "$0")/
+TESTDIR=$CURDIR/../..
+NUM_TESTS=18
+TEST_APP="$TESTDIR/utils/testapp/gen-syscall-events-callstack/gen-syscall-events-callstack"
+PARSE_CALLSTACK="$TESTDIR/utils/parse-callstack.py"
+
+SESSION_NAME="callstack"
+CHANNEL_NAME="chan0"
+
+source "$TESTDIR/utils/utils.sh"
+
+function lttng_untrack_all()
+{
+	lttng untrack -s "$SESSION_NAME" --all --pid -k > /dev/null
+}
+
+function lttng_track_pid()
+{
+	local PID=$1
+	lttng track -s "$SESSION_NAME" -k --pid="$PID" > /dev/null
```

Comment:
See this diff to make sure this works when the client is not installed (the utils in `utils.sh` launch the client from the `builddir`).

```
diff --git a/tests/regression/kernel/test_callstack b/tests/regression/kernel/test_callstack
index 5ffa33818..056acce26 100755
--- a/tests/regression/kernel/test_callstack
+++ b/tests/regression/kernel/test_callstack
@@ -19,7 +19,7 @@ TEST_DESC="Kernel tracer - Callstack context"
 
 CURDIR=$(dirname "$0")/
 TESTDIR=$CURDIR/../..
-NUM_TESTS=18
+NUM_TESTS=22
 TEST_APP="$TESTDIR/utils/testapp/gen-syscall-events-callstack/gen-syscall-events-callstack"
 PARSE_CALLSTACK="$TESTDIR/utils/parse-callstack.py"
 
@@ -30,13 +30,13 @@ source "$TESTDIR/utils/utils.sh"
 
 function lttng_untrack_all()
 {
-	lttng untrack -s "$SESSION_NAME" --all --pid -k > /dev/null
+	lttng_untrack 0 "-s $SESSION_NAME --all --pid -k"
 }
 
 function lttng_track_pid()
 {
 	local PID=$1
-	lttng track -s "$SESSION_NAME" -k --pid="$PID" > /dev/null
+	lttng_track 0 "-s $SESSION_NAME -k --pid=$PID"
 }
 
 function run_workload()
```

Reply:
done.